### PR TITLE
Remove eval from tool invocation

### PR DESCRIPTION
**Pull request by kgang**

### CODE REVIEW

File: concrete/operators/abstract.py [hunk 1]
```diff
@@ -47,7 +47,7 @@ def __new__(cls, clsname, bases, attrs):
         def _delay_factory(func: Callable[..., str]) -> Callable[..., AsyncResult]:
 
             def _delay(
-                self: AbstractOperator,
+                self: "AbstractOperator",
```
Comment (MichaelDeng03) [inline concrete/operators/abstract.py:50]: what was wrong with AbstractOperator?

Comment (kgang) [reply concrete/operators/abstract.py:50]: AbstractOperator as a class hasn't been defined yet when this runs

**Review by MichaelDeng03 — APPROVED** [review]
lgtm

File: concrete/operators/abstract.py [hunk 2]
```diff
@@ -134,7 +135,7 @@ def _qna(
         except AttributeError:
             pass
 
-        return response.content
+        return response_format(**json.loads(response.content))
```
Comment (MichaelDeng03) [inline concrete/operators/abstract.py:138]: very nice - you might also be able to use pydantic model_validate

Comment (kgang) [reply concrete/operators/abstract.py]: good call [](edited-with-Axolo)

File: concrete/orchestrator.py [hunk 3]
```diff
@@ -103,7 +103,7 @@ async def _do_work_plain(self) -> AsyncGenerator[tuple[str, str], None]:
                 tools=[AwsTool],
                 response_format=Tool,
             )
-            invoke_tool(**json.loads(deploy_tool_call))  # nosec
+            invoke_tool(**deploy_tool_call.__dict__)  # nosec
```
Comment (MichaelDeng03) [inline concrete/orchestrator.py:106]: Pydantic BaseModels implements .dict(). __dict__ includes a lot of BaseModel internals

**Pull request merged**